static/app/components/useButtonHook.tsx

```diff
@@ -0,0 +1,60 @@
+import {useCallback} from 'react';
+
+import HookStore from 'sentry/stores/hookStore';
+
+type Props = {
+  busy?: boolean;
+  disabled?: boolean;
+  onClick?: (e: React.MouseEvent) => void;
+};
+
+const useGetHook = () => {
+  const useButtonTracking = HookStore.get('react-hook:track-buttons')[0];
+  const useButtonClick = ({disabled, busy, onClick}: Props) =>
```

Comment:
I think you can make this logic a bit simpler and with less duplication. Something like:

```ts
const useButtonTracking = HookStore.get('react-hook:track-buttons')[0];

const handleButtonTracking = useButtonTracking?.(buttonProps)

const handleClick = (buttonProps: ButtonProps) =>
    useCallback(
      (e: React.MouseEvent) => {
        // Don't allow clicks when disabled or busy
        if (disabled || busy) {
          e.preventDefault();
          e.stopPropagation();
          return;
        }

        if (typeof onClick !== 'function') {
          return;
        }

        handleButtonTracking?.(buttonProps)

        onClick(e);
      },
      [disabled, busy, onClick]
    );

return handleClick
```

Then in the getsentry PR you can get rid of the guards that check for disabled, busy, and a click handler.

Reply:
great idea! i think this simplifies the code a lot.